### PR TITLE
feat: add warning for non vector embeddings

### PR DESCRIPTION
**Pull request by samsja**
Goals : warning user when instantiating a Document with an embedding which is not a vector (i.e one dimension ndarray)

fix : https://github.com/jina-ai/docarray/issues/351

### CODE REVIEW

File: docarray/document/mixins/property.py [hunk 1]
```diff
@@ -87,6 +89,16 @@ def matches(self, value: 'DocumentArray'):
 
         self._data.matches = value
 
+    @_PropertyMixin.embedding.setter
+    def embedding(self, value: 'ArrayType'):
+
+        if is_vector(value):
```
Comment (JoanFM) [inline docarray/document/mixins/property.py:95]: this can be a little overkill for every document.

**Review by JoanFM — CHANGES_REQUESTED** [review]
I am thinking maybe we should not do this

**Review by JoanFM — COMMENTED** [review]
Can we please benchmark the code?

File: docarray/document/mixins/property.py (hunk 1 above)
Comment (samsja) [reply docarray/document/mixins/property.py:95]: Yes I agree

**Comment by samsja** [comment]
> 
Here is some benchmark, the feature is not fully implemented yet (still a bug when creating an empty array) but before I go further and fix all the test we can decide if this overhead ( + 35%) is affordable.

Screenshot from 2022-05-25 16-19-49

@alaeddine-13 @JoanFM @davidbp

I would say that the difference is too big and we should not warn here

**Comment by samsja** [comment]
Conclusion : the overhead is not acceptable ( + 35%) on Document creation, therefore we drop the warning

**Pull request closed without merging**